transport-native-io_uring/src/test/java/io/netty/incubator/channel/uring/IOUringEventLoopTest.java

```diff
@@ -56,6 +58,16 @@ public void shutdownGracefullyZeroQuietBeforeStart() throws Exception {
         assertTrue(group.shutdownGracefully(0L, 2L, TimeUnit.SECONDS).await(1500L));
     }
 
+    @RepeatedTest(100)
+    public void shutdownNotSoGracefully() throws Exception {
+        // Test for race conditions to ensure we don't have race conditions that prevent a loop from shutting down.
+        EventLoopGroup group = newEventLoopGroup();
+        CountDownLatch latch = new CountDownLatch(1);
+        group.submit(() -> latch.countDown());
```

Comment:
Does this test really catches the race? It creates a single-threaded event-loop group, so I'd think that the task submitted here would always be executed after the event loop has started.

Reply:
Yes, it's what I used in https://github.com/netty/netty-incubator-transport-io_uring/pull/226 to debug. The task is run, and I think having a task available is what causes `hasTasks()` to evaluate to `true` and thus skip the `submissionQueue.submitAndWait();` call.

Reply:
Aha, I see. 👍